node.go

```diff
@@ -19,6 +20,7 @@ type Node struct {
 	LastHealthCheck time.Time
 
 	client *memcache.Client
+	lock   sync.RWMutex
```

Comment:
I suggest commenting on which members the lock is protecting, and making those private.